br/pkg/lightning/backend/external/iter.go

```diff
@@ -179,6 +182,15 @@ func newMergeIter[
 			elem:      e,
 			readerIdx: j,
 		})
+		sampleKeySize += len(e.sortKey())
+		sampleKeyCnt++
+	}
+	// We check the hotspot when the elements size is almost the same as the concurrent reader buffer size.
+	// So that we don't drop too many bytes if the hotspot shifts to other files.
+	if sampleKeySize == 0 {
```

Comment:
it's a bit safer to check `sampleKeySize/sampleKeyCnt` is zero

Reply:
If sampleKeySize > 0, then `sampleKeySize/sampleKeyCnt` can't be 0